### PR TITLE
[Runtime Env]Add usage of expansion of certain PATH variables in env_vars

### PR DESCRIPTION
**Pull request by larrylian**
## Why are these changes needed?

Add usage of expansion of certain PATH variables in env vars to documentation

## Related issue number

#34394 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/ray-core/handling-dependencies.rst [hunk 1]
```diff
@@ -367,10 +367,15 @@ The ``runtime_env`` is a Python dictionary or a Python class :class:`ray.runtime
   - Example: ``"pytorch_p36"``
 
 - ``env_vars`` (Dict[str, str]): Environment variables to set.  Environment variables already set on the cluster will still be visible to the Ray workers; so there is
-  no need to include ``os.environ`` or similar in the ``env_vars`` field.
+  no need to include ``os.environ`` or similar in the ``env_vars`` field.   
+  By default, these environment variables will override the same name environment variables on the cluster. 
+  If the user wants to append his value to the existing environment variable such as "PATH/LD_LIBRARY_PATH". he can refer to the example to set it.
```
Comment (jjyao) [inline doc/source/ray-core/handling-dependencies.rst:372]: You can also reference the existing environment variable via `${ENV_VAR}` and achieve the append behavior (currently only PATH,LD_LIBRARY_PATH,DYLD_LIBRARY_PATH,LD_PRELOAD are supported). See below for an example.

File: doc/source/ray-core/handling-dependencies.rst [hunk 2]
```diff
@@ -367,10 +367,15 @@ The ``runtime_env`` is a Python dictionary or a Python class :class:`ray.runtime
   - Example: ``"pytorch_p36"``
 
 - ``env_vars`` (Dict[str, str]): Environment variables to set.  Environment variables already set on the cluster will still be visible to the Ray workers; so there is
-  no need to include ``os.environ`` or similar in the ``env_vars`` field.
+  no need to include ``os.environ`` or similar in the ``env_vars`` field.   
+  By default, these environment variables will override the same name environment variables on the cluster.
```
Comment (angelinalg) [inline doc/source/ray-core/handling-dependencies.rst:371]:
```suggestion
  By default, these environment variables override the same name environment variables on the cluster. 
```

File: doc/source/ray-core/handling-dependencies.rst [hunk 3]
```diff
@@ -367,10 +367,15 @@ The ``runtime_env`` is a Python dictionary or a Python class :class:`ray.runtime
   - Example: ``"pytorch_p36"``
 
 - ``env_vars`` (Dict[str, str]): Environment variables to set.  Environment variables already set on the cluster will still be visible to the Ray workers; so there is
-  no need to include ``os.environ`` or similar in the ``env_vars`` field.
+  no need to include ``os.environ`` or similar in the ``env_vars`` field.   
+  By default, these environment variables will override the same name environment variables on the cluster. 
+  You can also reference existing environment variables using ${ENV_VAR} and achieve the append behavior.
```
Comment (angelinalg) [inline doc/source/ray-core/handling-dependencies.rst:372]:
```suggestion
  You can also reference existing environment variables using ${ENV_VAR} to achieve the appending behavior.
```

File: doc/source/ray-core/handling-dependencies.rst [hunk 4]
```diff
@@ -367,10 +367,15 @@ The ``runtime_env`` is a Python dictionary or a Python class :class:`ray.runtime
...
+  Currently, only PATH, LD_LIBRARY_PATH, DYLD_LIBRARY_PATH, and LD_PRELOAD are supported. See below for an example:
```
Comment (angelinalg) [inline doc/source/ray-core/handling-dependencies.rst:373]:
```suggestion
  Only PATH, LD_LIBRARY_PATH, DYLD_LIBRARY_PATH, and LD_PRELOAD are supported. See below for an example:
```

**Review by angelinalg — APPROVED** [review]
LGTM

**Pull request merged**